### PR TITLE
Environment for control of training convergence

### PR DESCRIPTION
**Pull request by iaroslav-ai**
These commits contain an environment where agent can adjust parameters like step size, momentum etc during training of neural net to improve convergence / quality of end - result. 

One episode is a training of one neural net for 20 epochs. Agent can adjust parameters in the beginning of every epoch. Reward is proportional to the best validation accuracy, engineered in a way to encourage trade off between convergence speed and accuracy.

The parameters that agent can adjust are learning rate, momentum, batch size, l1 and l2 penalty coefficients. As a feedback, agent receives # of instances / labels in dataset, description of network architecture, and validation accuracy for every epoch.

Architecture of neural network and dataset used are selected randomly at the beginning of episode. Datasets used are MNIST, CIFAR10, CIFAR100. Network architectures contain multilayer convnets 66 % of the time, and are [classic] feedforward nets otherwise.

Number of instances in datasets are chosen at random in range from around 100% to 10% such that adjustment of l1, l2 penalty coefficients makes more difference.

I came up with this environment during preparation of environment as discussed in #94 . This environment requires Keras and Theano or TensorFlow to run. It is best to run it on gpu.

Please let me know if this is interesting to you. You can run the environment using committed "test_tuning.py".

### CODE REVIEW

**Comment by jietang** [comment]
Hi @iaroslav-ai, this looks great and I'd love to merge it. Could you make the following changes:
1) add the Python dependencies (e.g. keras) to setup.py (you can make a new category say parameter_tuning) 
2) add the task to the scoreboard registry (gym/scoreboard/__init__.py) so it shows up on the website with the experimental=True tag
3) add an exception in gym/envs/tests/test_envs.py that prevents instantiating the ConvergenceControl environment when running nose tests (otherwise keras will pull in cifar, etc. during CI)

**Comment by iaroslav-ai** [comment]
Hi @jietang. I added changes according to your comment, please let me know if something is missing.

**Comment by iaroslav-ai** [comment]
If everything is ok, please also include commits that I made recently to the pull request (these are small fixes / improvement to environment)

**Comment by jietang** [comment]
Thanks @iaroslav-ai
